ppb/sprites.py

```diff
@@ -141,18 +141,44 @@ def _attribute_gate(self, attribute, bad_sides):
 
 
 class BaseSprite(EventMixin):
-
+    """
+    The base Sprite class. All sprites should inherit from this (directly or
+    indirectly).
+
+    Attributes:
+    * image (str): The image file
+    * resource_path (pathlib.Path): The path that image is relative to
+    * position: Location of the sprite
+    * facing: The direction of the "top" of the sprite (rendering only)
+    * size: The width/height of the sprite (sprites are square)
+    """
     image = None
     resource_path = None
+    position: Vector = Vector(0, 0)
+    facing: Vector = Vector(0, -1)
+    _size: Union[int, float] = 1
+    _offset_value = None
 
-    def __init__(self, size: Union[int, float]=1, pos: Iterable=(0, 0),
-                 facing: Vector=Vector(0, -1)):
+    def __init__(self, **kwargs):
         super().__init__()
-        self.position = Vector(*pos)
-        self._offset_value = None
-        self._size = None
-        self.size = size
-        self.facing = facing
+
+        # Make these instance properties with fresh instances
+        # Don't use Vector.convert() because we need copying
+        self.position = Vector(*self.position)
+        self.facing = Vector(*self.facing)
+
+        # Initialize things
+        for k, v in kwargs.items():
+            # Abbreviations
+            if k == 'pos':
+                k = 'position'
+            # Castings
+            if k in ('position', 'facing'):
+                v = Vector(*v)  # Vector.convert() when that ships.
+            setattr(self, k, v)
+
+        # Trigger some calculations
+        self.size = self.size
```

Comment:
You are memoizing `offset_value`. That's to cover the case if `size` isn't assigned above, so that the `size` setter calculations are run without actually changing the value.

The only other way I know how to handle this would be to duplicate code or add an extra method just to handle the calculation.

Yes, this is super weird and possibly deserving of a more in-depth comment, but I'm not sure the alternatives are actually better.